code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -372,41 +372,51 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 /datum/materialProc/molitz_temp
 	max_generations = 1
 	execute(var/atom/owner, var/temp, var/agent_b=FALSE)
-		if(!istype(owner.material, /datum/material/crystal/molitz))
-			return
+		if(temp < 500) return //less than reaction temp
+
 		var/datum/material/crystal/molitz/molitz = owner.material
-		var/turf/target = get_turf(owner)
+		if(!istype(molitz)) CRASH("Molitz_temp material proc applied to non-molitz thing") //somehow applied to non-molitz
+
 		if(molitz.iterations <= 0) return
-		if(ON_COOLDOWN(owner, "molitz_gas_generate", 30 SECONDS)) return
 
-		var/datum/gas_mixture/air = target?.return_air()
-		if(!air) return
+		var/datum/gas_mixture/air
+		if(hasvar(owner, "air_contents"))
+			air = owner:air_contents
+		if(!air && hasvar(owner.loc, "air_contents"))
+			air = owner.loc:air_contents
```

Comment:
Can this be limited to `/obj/machinery/atmospherics` or what all you hoping to capture with this (or anything and everything that... could)?

Reply:
Reactor components don't subclass `/obj/machinery/atmospherics`, but do have an `air_contents` var. I mostly went with this approach because it'll work with anything that also supports using the atmospheric analyser.